x/emissions/module/network_summaries.go

```diff
@@ -132,10 +117,132 @@ func CalcForcastImpliedInferencesAtTime(
 				weightInferenceDotProduct += w_ijk * float64(inferenceByWorker[forecast.ForecastElements[j].Inferer].Value.Uint64())
 				weightSum += w_ijk
 			}
-			I_i[k] = I_i[k] / weightSum
+			I_i[k] = emissions.Inference{
+				Worker: forecast.Forecaster,
+				Value:  cosmosMath.NewUint(uint64(weightInferenceDotProduct / weightSum)),
+			}
 		}
 	}
 
 	// A value of 0 => no inference corresponded to any of the forecasts from a forecaster
 	return I_i, nil
 }
+
+// Calculate the forecast-implied inferences at a given time using eq3-8 from the litepaper,
+// inferences and forecasts from the worker update at or just after the given time, and
+// losses from the loss update just before the given time.
+func GetAndCalcForcastImpliedInferencesAtTime(
+	ctx sdk.Context,
+	k keeper.Keeper,
+	topicId TopicId,
+	time uint64) ([]emissions.Inference, error) {
+	// Get inferences from worker update at or just after the given time
+	inferences, err := k.GetInferencesAtOrAfterTime(ctx, topicId, time)
+	if err != nil {
+		fmt.Println("Error getting inferences: ", err)
+		return nil, err
+	}
+	// Get forecasts from worker update at or just after the given time
+	forecasts, err := k.GetForecastsAtOrAfterTime(ctx, topicId, time)
+	if err != nil {
+		fmt.Println("Error getting forecasts: ", err)
+		return nil, err
+	}
+	// Get losses from loss update just before the given time
+	networkLossBundle, err := k.GetNetworkLossBundleAtOrBeforeTime(ctx, topicId, time)
+	if err != nil {
+		fmt.Println("Error getting network losses: ", err)
+		return nil, err
+	}
+
+	return CalcForcastImpliedInferencesAtTime(ctx, k, topicId, inferences, forecasts, networkLossBundle)
+}
+
+// Calculates eq9-11 from the litepaper using the given inferences, forecast-implied inferences, and network regrets
```

Comment:
I think you should never refer to these equations by number because the number of an equation in the paper may change based on some edits to the paper over time. In my PRs I copy the actual equation math and put it directly in the comment. That way folks referencing the paper after the fact can look for the equation in the paper for the relevant information without any possibility that equation 9-11 may be totally different equations in the paper 3 months from now

It also makes my job easier as your PR reviewer, I don't have to go look at the paper to figure out wtf equations 9-11 are to check your implementation of them isn't broken

Reply:
done